### ISSUE TITLE
mongodb OOM crash after flooding it with events

### ISSUE THREAD

**Issue by christian-kreuzberger-dtx** [keptn/keptn#3968]
With Keptn 0.8.2 and dynatrace-sli-service, I executed the following command (spamming evaluation events) for about 2 hours and my mongodb crashed multiple times due to Out Of Memory (OOM) errors:
```
while true; do keptn trigger evaluation --project=sockshop --stage=staging --service=carts  --timeframe=5m; sleep 5; done;
```
![image](https://user-images.githubusercontent.com/56065213/117115451-6ca9e680-ad8d-11eb-9156-82732603ed93.png)
![image](https://user-images.githubusercontent.com/56065213/117116576-cf4fb200-ad8e-11eb-8c9a-c8e48651d561.png)

I'm not convinced that there is a memory leak in mongodb though, I believe the root-cause of this is remediation-service opening connections to mongodb and not closing it, as described in https://github.com/keptn/keptn/issues/3936

According to @warber , we are removing the mongodb connection from remediation-service in the upcoming release, so this should not be a problem any longer.

# Proposed actions

- Increase memory limits for mongodb (needs discussion)
- Add SLIs/SLOs to our keptn self monitoring such that we can detect that mongodb crashed (e.g., memory usage)
- Streamline mongodb connection for our microservices, such that we never run into this problem again

**Comment by warber** [comment]
To be retested with Keptn 0.8.3

**Comment by christian-kreuzberger-dtx** [comment]
Seems like this was resolved using Keptn 0.8.3.
